iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/helpers/IotHubServicesCommon.java

```diff
@@ -23,6 +23,11 @@
  */
 public class IotHubServicesCommon
 {
+    protected static final Integer RETRY_MILLISECONDS = 100;
+    protected static final Integer SEND_TIMEOUT_MILLISECONDS = 60000;
+
+    public static final int ERROR_INJECTION_MESSAGE_TIMEOUT_MILLISECONDS = 200;
```

Comment:
We set a short expiry time on these fault injection messages because they often fail to be acknowledged by the service before the connection is lost. With this expiry time, our retry logic wont send this message more than a few times. At the same time though, we risk them never being sent. That's why I added some retry logic to prevent that

Reply:
What would cause the service to drop the connection before ack'ing the receipt of this fault injection request? Afaik, service will drop the connection on the next message that they receive (after the FI request message). So this should mean that the FI request message would always be ack'ed.
In java, do we not wait until message 1 is ack'ed, before sending message 2? Is that because of the way the send message queue works?

Reply:
> Afaik, service will drop the connection on the next message that they receive (after the FI request message)

That hasn't been my experience. The Java tests don't even send another message until after the connection is dropped. From my experience, the service may or may not ack the fault injection message before dropping the connection.

Reply:
I'll spend some time investigating if we can rely on the service ack'ing these fault injection messages before merging this in